scripts/install/install-dependencies.sh

```diff
@@ -0,0 +1,394 @@
+function build_cfitsio() {
+  echo
+  echo "Building cfitsio..."
+  echo
+  cd "${PANDIR}/tmp/cfitsio"
+  make
+  make utils stand_alone shared
+}
+
+# Print the version of the downloaded and built cfitsio version number, as in
+# its pkg-config file (cfitsio.pc).
+function latest_cfitsio_version() {
+  extract_version_from_pkg_config tmp/cfitsio/cfitsio.pc
+}
+
+# Prints the installed cfitsio version number.
+function installed_cfitsio_version() {
+  pkg-config --mod-version --silence-errors cfitsio || /bin/true
+}
+
+# Install the downloaded and built cfitsio.
+function sudo_install_cfitsio() {
+  echo
+  echo "Running sudo make install for cfitsio, you may be prompted for your password."
+  echo
+  cd "${PANDIR}/tmp/cfitsio"
+  (set -x ; sudo make install)
+}
+
+# Install the latest versio of cfitsio if it is not already installed.
+function install_latest_cfitsio() {
+  (fetch_and_configure_cfitsio)
+  local -r LATEST_CFITSIO_VERSION="$(latest_cfitsio_version)"
+  local -r INSTALLED_CFITSIO_VERSION="$(installed_cfitsio_version)"
+  if [[ "${LATEST_CFITSIO_VERSION}" == "${INSTALLED_CFITSIO_VERSION}" ]] ; then
+    echo "Installed cfitsio is at latest version (${LATEST_CFITSIO_VERSION})."
+  else
+    if [[ -z "${INSTALLED_CFITSIO_VERSION}" ]] ; then
+      echo "Installing cfitsio version ${LATEST_CFITSIO_VERSION}."
+    else
+      echo "Installing cfitsio version ${LATEST_CFITSIO_VERSION}, replacing" \
+        "installed version ${OLD_CFITSIO_VERSION}."
+    fi
+    (build_cfitsio)
+    (sudo_install_cfitsio)
+  fi
+  rm -rf tmp/cfitsio_latest.tar.gz tmp/cfitsio
+}
+
+# Downloads astrometry version ${ASTROMETRY_VERSION} into a temp directory,
+# then builds and installs into ${PANDIR}/astrometry.
+# TODO(jamessynge): Discuss whether to build directly in ${PANDIR}/astrometry,
+# or instead installing into /usr/local as we do with cfitsio.
+function install_astrometry() {
+  local -r DIR="astrometry.net-${ASTROMETRY_VERSION}"
+  local -r FN="${DIR}.tar.gz"
+  local -r URL="http://astrometry.net/downloads/${FN}"
+  local -r SCRATCH_DIR="$(mktemp -d "${TMPDIR:-/tmp/}install-astrometry.XXXXXXXXXXXX")"
+  local -r INSTALL_TO="${PANDIR}/astrometry"
+  cd "${SCRATCH_DIR}"
+
+  echo_bar
+  echo  
+  echo "Fetching astrometry into directory $(pwd)"
+
+  wget "${URL}"
+  tar zxvf "${FN}"
+  cd "${DIR}"
+
+  echo "Building ${DIR}"
+  make
+  make py
+  if [[ -d "${INSTALL_TO}" ]] ; then
+    rm -rf "${INSTALL_TO}"
+  fi
+  mkdir -p "${INSTALL_TO}"
+  echo "Installing into ${INSTALL_TO}"
+  make install "INSTALL_DIR=${INSTALL_TO}"
+
+  add_to_PATH "${INSTALL_TO}/bin"
+  cd "${PANDIR}"
+  rm -rf "${SCRATCH_DIR}"
+}
+
+function install_astrometry_indices() {
+  mkdir -p "${PANDIR}/astrometry/data"
+  cd "${PANDIR}/astrometry/data"
+  # See http://astrometry.net/doc/readme.html#getting-index-files
+  # for info on the index files you might need.
+  if [[ "${DO_ASTROMETRY_FULL_INDICES}" -eq 1 ]] ; then
+    if [[ ! -f index-4107.fits ]] ; then
```

Comment:
Just a note that if download/install gets interrupted this will not work properly second time around.

We also have `$POCS/pocs/utils/data.py` that does this same checking and is run before tests.

Reply:
I notice that it installs more index files.

Do you have a suggestion regarding the interruption? Explicitly do the download of each?
Or just run $POCS/pocs/utils/data.py near the end of the install.

FWIW, I'd like to eliminate all downloading from tests.

Reply:
The tests shouldn't download but they rely on having the files previously downloaded. As of right now they download [here](https://github.com/panoptes/POCS/blob/develop/pocs/tests/conftest.py#L9).

The astroplan IRES stuff still needs to be figured out better as per #121